lib/doofinder_api_items.php

```diff
@@ -55,6 +60,24 @@ public function deleteBulk($payload)
         return $this->post($url, $payload);
     }
 
+    /**
+     * Make a request to the API to reprocess all the feeds
+     *
+     * @param array items ids
+     */
+    public function invokeReindexing()
+    {
+        $installation_id = Configuration::get('DF_INSTALLATION_ID');
+        $json_data = json_encode(['query' => 'mutation { process_store_feeds(id: "' . $installation_id . '", callback_url: "' . $this->getProcessCallbackUrl() . '") { id }}']);
+        $response = $this->post($this->api_url . '/api/v1/graphql.json', $json_data);
+        if (empty($response)) {
+            return;
+        }
+        if (empty($response_array['errors'])) {
+            Configuration::updateValue('DF_FEED_INDEXED', false);
+        }
```

Comment:
This logic has to be out.
This class should be for connecting to the API only.

Reply:
Totally agree, changed on: https://github.com/doofinder/doofinder-prestashop/pull/143/commits/13a4e4631a3fff59967320600ca685099550cd59